### PR TITLE
docs(docs): add native versioning

### PR DESCRIPTION
**Pull request by yoonhyejin**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

---

This should be merged first: https://github.com/datahub-project/static-assets/pull/48
Docs release process runbook : https://www.notion.so/acryldata/Docs-Release-Process-_V1-6ef51f4a98d74f4c8d92069077425589?pvs=4

### CODE REVIEW

File: docs-website/build.gradle [hunk 1]
```diff
@@ -77,7 +77,12 @@ task yarnGenerate(type: YarnTask, dependsOn: [yarnInstall,
   args = ['run', 'generate']
 }
 
-task yarnStart(type: YarnTask, dependsOn: [yarnInstall, yarnGenerate]) {
+task downloadHistoricalVersion(type: Exec) {
```
Comment (hsheth2) [inline docs-website/build.gradle:80]:
```suggestion
task downloadHistoricalVersions(type: Exec) {
```

File: docs-website/build.gradle [hunk 2]
```diff
@@ -77,7 +77,12 @@ task yarnGenerate(type: YarnTask, dependsOn: [yarnInstall,
...
+    workingDir '.'
+    commandLine 'python', 'download_historical_versions.py'
```
Comment (hsheth2) [inline docs-website/build.gradle:82]:
```suggestion
    commandLine 'python3', 'download_historical_versions.py'
```

File: docs-website/build.gradle [hunk 3]
```diff
@@ -105,8 +110,7 @@ task serve(type: YarnTask, dependsOn: [yarnInstall] ) {
 }
 
 
-task yarnBuild(type: YarnTask, dependsOn: [yarnLint, yarnGenerate]) {
-  inputs.files(projectMdFiles)
+task yarnBuild(type: YarnTask, dependsOn: [yarnLint, yarnGenerate, downloadHistoricalVersion]) {  inputs.files(projectMdFiles)
```
Comment (hsheth2) [inline docs-website/build.gradle:113]: put the inputs on a newline?

File: docs-website/download_historical_versions.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    data = response.json()
+    tar_urls = [
+        file["download_url"] for file in data if file["name"].endswith(".tar.gz")
+    ]
+    print(tar_urls)
+    return tar_urls
+
+
+def main():
+    repo_url = "https://api.github.com/repos/acryldata/static-assets-test"
```
Comment (hsheth2) [inline docs-website/download_historical_versions.py:28]: change to datahub project's static assets repo

also ideally put this in a constant at the top

File: docs-website/versioned_sidebars/version-0.10.5-sidebars.json
Comment (hsheth2) [inline docs-website/versioned_sidebars/version-0.10.5-sidebars.json:1]: can we include these sidebars in the static-assets tar.gz files, and then make the whole `versioned_sidebars` directory generated at build time?

File: docs-website/src/pages/docs/_components/SearchBar/index.jsx
Comment (hsheth2) [inline docs-website/src/pages/docs/_components/SearchBar/index.jsx:1]: hard to follow - which of these were formatting changes vs actual changes?

File: docs-website/download_historical_versions.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+    for url in tar_urls:
+        filename = os.path.basename(url)
+        destination_path = os.path.join(destination_dir, filename)
+
+        try:
+            download_file(url, destination_path)
+            print(f"Downloaded {filename} to {destination_dir}")
+            with tarfile.open(destination_path, "r:gz") as tar:
+                tar.extractall()
```
Comment (hsheth2) [inline docs-website/download_historical_versions.py:45]: we could also add a check that avoids re-downloading from the static assets targz if it's already been downloaded and extracted

**Review by hsheth2 — COMMENTED** [review]
I'm super excited about this

File: docs-website/src/pages/docs/_components/SearchBar/index.jsx
Comment (yoonhyejin) [reply docs-website/src/pages/docs/_components/SearchBar/index.jsx:1]: reverted back all the formatting changes here

**Review by hsheth2 — APPROVED** [review]
